cl/lib/search_utils.py

```diff
@@ -1031,6 +1035,36 @@ def add_depth_counts(
         return None
 
 
+def get_parenthetical_groups_with_cache(
+    cluster: OpinionCluster,
+) -> List[ParentheticalGroup]:
+    """
+    Given an opinion cluster, get its parentheticals and categorize them
+    into groups based on textual similarity.
+
+    @param cluster: The cluster to fetch and group parentheticals for
+    @return: A list of ParentheticalGroup's for the given cluster
+    """
+
+    cache_key = f"parenthetical_groups:{cluster.pk}"
+    cache = caches["db_cache"]
+    cached_groups, num_parentheticals = cache.get(cache_key, (None, None))
+    if (
+        cached_groups is not None
+        # Groups will be recomputed when new parentheticals are added
+        and num_parentheticals == cluster.parentheticals.count()
+    ):
+        return cached_groups
+
+    # If the case has more than 1,000 parentheticals, only use the top 1,000
+    parentheticals = list(cluster.parentheticals[:1000])
+    groups = get_parenthetical_groups(parentheticals)
+    a_week = 60 * 60 * 24 * 7
+    cache.set(cache_key, (groups, len(parentheticals)), a_week)
```

Comment:
This needs to use len(parentheticals.count()) right? Otherwise, it will always say 1,000 for big cases? 
```
    cache.set(cache_key, (groups, len(parentheticals)), a_week)
```

Reply:
True, good catch.